test/question.test.js

```diff
@@ -1,67 +1,135 @@
-    //             expect(resp_data).to.be.an('object');
-    //             expect(resp_data).to.have.property('user');
-    //             expect(resp_data.user.id).to.be.an('number');
-    //         });
-    //     });
+            const resp_data = response.body;
+            expect(resp_data).to.be.an('object');
+            expect(resp_data).to.have.property('data');
+            expect(resp_data.data).to.be.an('array');
+        });
+    });
 
+    //negative test for post question
     describe('Negative Tests', function () {
         it('should not post a question', async function () {
             const response = await supertest(app)
-                .post('/api/v1/signup')
+                .post('/api/v1/question')
+                .set({
+                    Authorization: sample_data.auth_token,
+                    Accept: 'application/json',
+                })
                 .send({
-                    question_text: 'My first age',
                     user_id: '24',
                 })
-                .set('Accept', 'application/json')
-                .expect(409);
+                .expect(400);
+            const resp_data = response.body;
+            expect(resp_data).to.be.an('object');
+            expect(resp_data.messages).to.be.an('string');
+            expect(resp_data.messages).to.contain('Question text is required');
+        });
+    });
 
+    //negative test for delete question
+    describe('Negative Tests', function () {
+        it('should not delete a question', async function () {
+            const response = await supertest(app)
+                .delete('/api/v1/question/:id')
+                .set({
+                    Authorization: sample_data.auth_token,
+                    Accept: 'application/json',
+                })
+                .send({
+                    id: '900',
+                })
+                .expect(404);
             const resp_data = response.body;
             expect(resp_data).to.be.an('object');
             expect(resp_data.message).to.be.an('string');
-            expect(resp_data.message).to.contain('Email already taken');
+            expect(resp_data.message).to.contain('No question with such id');
+        });
+    });
+
+    //negative test for get question
+    describe('Negative Tests', function () {
+        it('should not successfully get a question', async function () {
+            const response = await supertest(app)
+                .get('/api/v1/question')
+                .set({
+                    Authorization: sample_data.auth_token,
+                    Accept: 'application/json',
+                })
+                .send({
+                    user_id: '1',
+                })
+                .expect(200);
+
+            const resp_data = response.body;
+            expect(resp_data).to.be.an('object');
+            expect(resp_data).to.have.property('data');
+            expect(resp_data.data).to.be.an('array');
         });
```

Comment:
I don't think this test case is necessary, because it is similar to the first test case on line 56 of this file. What did you wanted to test here? 🤔

Reply:
This is a negative test, while the one above is the positive test.

Reply:
Well, negative tests are for failure points, that is, you test and confirm that the application actually failed when certain input or condition is met as stipulated in the codes and also that the expected error message was returned. However, this is not the case here, the only different between this and the other one is just the test-case label `should not successfully get a question` but your assertion expect it to have `data` as an array of questions. So going through your post question controller there is no explicit failure point that can be tested effectively, that was the reason I mark that test case as unnecessary. 

For example, when testing post questions, there was no failure point in your post question controller initially and I recommended adding some lines of codes to check that the user must supply `question_text` else throw an error. You added a negative test that assert that `Question text is required` is returned as an error if `question_text` is not supplied when calling that API, so that was a failure point. So it has to be tested to ensure that the application conforms to that expectation, that is, if `question_text` is not supplied, the application actually threw the expected error.

So relating this to the get questions api, you can see that there is no explicit place in that controller where error is returned aside the one inside the `try  catch` which would be difficult to test because creating a conducive environment or condition to trigger such error is not achievable.

I don't know if this makes sense or I've successfully complicated and compounded thing 🤔